From 79572c004853199174d68dda7057a058f06698f3 Mon Sep 17 00:00:00 2001
From: b02550 <b02550@freescale.com>
Date: Fri, 12 Jun 2009 14:35:04 +0800
Subject: [PATCH 062/256] ENGR00113242 IPUv3: Using new ioctl to get ipu
 channel type of fb.

Using new ioctl to get ipu channel type to avoid the conflict between
3Stack and BBG board.

Signed-off-by: DingQiang <b02550@freescale.com>
---
 ipu/mxc_ipu_hl_lib.c | 13 +++----------
 1 file changed, 3 insertions(+), 10 deletions(-)

diff --git a/ipu/mxc_ipu_hl_lib.c b/ipu/mxc_ipu_hl_lib.c
index c927996..947f627 100644
--- a/ipu/mxc_ipu_hl_lib.c
+++ b/ipu/mxc_ipu_hl_lib.c
@@ -954,16 +954,9 @@ static int _ipu_mem_alloc(ipu_lib_input_param_t * input,
 				goto err;
 			}
 
-			if (strcmp(fb_fix.id, "DISP3 FG") == 0)
-				ipu_priv_handle->output[j].fb_chan = MEM_FG_SYNC;
-			else if (strcmp(fb_fix.id, "DISP3 BG") == 0)
-				ipu_priv_handle->output[j].fb_chan = MEM_BG_SYNC;
-			else if (strcmp(fb_fix.id, "DISP3 BG - DI1") == 0)
-				ipu_priv_handle->output[j].fb_chan = MEM_DC_SYNC;
-
-			if (!ipu_priv_handle->output[j].fb_chan) {
-				dbg(DBG_WARNING,
-					"Get FB ipu channel failed, fix id %s\n", fb_fix.id);
+			if (ioctl(ipu_priv_handle->output[j].fd_fb, MXCFB_GET_FB_IPU_CHAN,
+				&ipu_priv_handle->output[j].fb_chan) < 0) {
+				dbg(DBG_WARNING,"Get FB ipu channel failed, use default\n");
 				if (output->fb_disp.fb_num == 0)
 					ipu_priv_handle->output[j].fb_chan = MEM_BG_SYNC;
 				else if (output->fb_disp.fb_num == 1)
-- 
1.8.0

